conda-store-server/conda_store_server/build.py

```diff
@@ -201,15 +207,24 @@ def build_conda_environment(db: Session, conda_store, build):
         build.size = context.result["disk_usage"]
 
         set_build_completed(db, conda_store, build)
+    # Always mark build as failed first since other functions may throw an
+    # exception
     except subprocess.CalledProcessError as e:
+        set_build_failed(db, build)
```

Comment:
Don't we want to add more info here too via `status_info`?

Reply:
No, see the comment right below this:

>     except BuildPathError as e:
>        # Provide status_info, which will be exposed to the user, ONLY in this
>        # case because the error message doesn't expose sensitive information

It's only done in specific cases where we control the error message (right now, it's only `BuildPathError` added in this PR). We don't pass random data here. See https://github.com/conda-incubator/conda-store/pull/653/files/34ac2dd4380dda1da3264e0482bafcd0e79976bf#r1390449347